### PR TITLE
Make strict mode == true raise an exception

### LINKED ISSUE #66: Strict mode is confusing
I would expect "strict true" to be the mode that raises exceptions rather than ignoring unhandled keys.

cc @dianacamacho

### PR DESCRIPTION
**Pull request by dpikt**
Resolves #66 and removes the `:with_exception` special key.

### CODE REVIEW

File: lib/decanter/configuration.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Configuration
     attr_accessor :strict
 
     def initialize
-      @strict = :with_exception
+      @strict = true
```
Comment (inveterateliterate) [inline lib/decanter/configuration.rb]: does the readme need to be updated or are you doing all of that in a later PR?

Comment (dpikt) [reply lib/decanter/configuration.rb]: I actually did it in a previous pr 😳 but I'm going to open a PR with a migration guide soon.

Comment (inveterateliterate) [reply lib/decanter/configuration.rb]: ah here: https://github.com/LaunchPadLab/decanter/pull/89/files

File: lib/decanter/core.rb [hunk 2]
```diff
@@ -102,19 +102,11 @@ def unhandled_keys(args)
...
-            {}
-          when :with_exception            
-            raise(UnhandledKeysError, "#{self.name} received unhandled keys: #{unhandled_keys.join(', ')}.")
-          else
-            args.select { |key| unhandled_keys.include? key }
-          end
-        else
-          {}
+        return {} unless unhandled_keys.any?
+        if strict_mode
```
Comment (inveterateliterate) [inline lib/decanter/core.rb]: ruby guard syntax? :grin: 

`raise(UnhandledKeysError, "#{self.name} received unhandled keys: #{unhandled_keys.join(', ')}.") if strict_mode `

Comment (dpikt) [reply lib/decanter/core.rb]: The line is so longgg

But yes I can do that.

**Pull request merged**